doc/pins/send_to_ingress_hld.md

```diff
@@ -0,0 +1,178 @@
+* [Open/Action items - if any](#open-action-items---if-any)
+
+## Revision
+
+Rev  | Rev Date   | Author(s)              | Change Description
+---- | ---------- | -----------------------| ------------------
+v0.1 | 10/25/2022 | Yilan Ji, Stephen Wang | Initial version
+
+## Scope
+
+Send to ingress feature in GPINs is to support [Packet I/O](https://github.com/sonic-net/SONiC/blob/master/doc/pins/Packet_io.md) transmit path that injects packets into the dataplane by the CPU. This doc is to extend the availability of ingress inject packet I/O to all applications in Packet I/O Transmit Path design.
+
+## Definitions/Abbreviations
+
+**PINS**: P4 Integrated Network Stack (PINS) provides a remote interface to SAI using P4.
+
+## Overview
+
+Packet I/O transmit involves taking care of 2 types of packets that have to be transmitted: Direct transmit and Ingress pipeline inject.
+
+### Direct transmit
+
+If the sender(applications like LACP or P4RT) already know the egress port that the packet should be sent out on, it is done by placing the packet on the socket interface of the corresponding netdev interface.
+
+### Ingress pipeline inject
+
+In the SDN scenario, the controller has the complete view of the whole network and programs the switch ASIC with the optimized routing rules that the switch NOS is not aware of. Also in some other scenario, that the ASIC has a better understanding of the current bandwidth/queue depth of each port that it can choose a better egress port from the ECMP/WCMP group. The sender application on the switch would like to inject the packet to the ASIC’s ingress pipeline to allow the switch ASIC to make the final decision.
+
+Injecting packets to the ingress pipeline can also be used as a way to test ASIC routing behavior in the standalone test environment.
+
+## Requirements
+
+* Provide configuration support of send to ingress in the port table.
+
+## Architecture Design
+
+![drawing](images/send_to_ing.png)
+
+Details of the transmit path in GPINs Packet I/O can be found in this [HLD](https://github.com/sonic-net/SONiC/blob/master/doc/pins/Packet_io.md#transmit-path).
+
+## High-Level Design
+
+This HLD introduces a new “Send To Ingress” port type in the port table that SWSS will create a corresponding netdev/host interface. Any application can inject the packet to the ASIC’s ingress pipeline and let the ASIC decide which port to send packet out based on the routing table and current state of the network.
+
+Note that the Send to Ingress host interface is only for packet out as it is only used in TX path, no packet ins are expected.
+
+### CONFIG DB schema
+
+Add knob in PORT table in CONFIG DB or config_db.json to enable send to ingress
+
+```
+  "PORT": {
+    "Ethernet0": {
+      "admin_status": "up",
+      "alias": "NoAliasEth0/1",
+      "index": "1",
+      "lanes": "9,10,11,12,13,14,15,16",
+      "mtu": "9100",
+      "speed": "400000"
+    },
+    …
+  },
+  "SEND_TO_INGRESS_PORT": {
+     "SEND_TO_INGRESS": {}
```

Comment:
Can you please provide the full info here? What is expected in the `{}`

Reply:
No attributes are needed currently. Only port name, `SEND_TO_INGRESS`, is needed as the configurable port name, and netdev name.